### ISSUE TITLE
investigate bun

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#60]
it might make pyright faster

**Comment by znd4** [comment]
Barely on-topic, but would it be easy to allow "bring your own pyright / node / bun"? If so, it'd probably be pretty easy to implement a solution for #177, although it'd require a bit of maintenance to keep the pyright version correct.

I'll try digging into the code to see if there's a light-touch way to implement this.

**Comment by KotlinIsland** [comment]
It wouldn't be possible to bring your own pyright, this project is setup such that it modifies a lot of source, not just acts as an 'expansion pack'.

**Comment by DetachHead** [comment]
i might add an option to byo node, which would hopefully fix these issues: #177 #172 #167

when i first started this project, i just wanted it to be a vscode extension where users could specify any version of pyright (the only issue i was trying to fix at the time was [the ability to pin the version used by vscode](https://github.com/DetachHead/basedpyright?tab=readme-ov-file#ability-to-pin-the-version-used-by-vscode)), but i literally couldn't get the language server to work without modifying how the pyright language server is built (c1c8be9d3975f8a643db4d4e0cf2e982d12011a3) so i gave up on that approach. i think even if we can get it working, it's not something worth supporting. basedpyright's additional functionality can be disabled to have essentially a backwards-compatible version of pyright, which i think covers that use case.

**Comment by jfcherng** [comment]
Just want to share my test on a Win11 x64 PC with recently Bun v1.1.0, which is the first release supporting Windows.

I don't "feel" (no numerical statistics, just my feeling) it's obviously faster, but it consumes much more memory than Electron / Node.js.

![image](https://github.com/DetachHead/basedpyright/assets/6594915/9966a79e-0a60-45f9-93cc-bf5c6f35fd6b)

Just an FYI, you can use Electron to run as node by setting env variable [`ELECTRON_RUN_AS_NODE=1`](https://www.electronjs.org/docs/latest/api/environment-variables#electron_run_as_node) and it consume less memory due to its [optimizations](https://www.electronjs.org/docs/latest/tutorial/performance).

**Comment by DetachHead** [comment]
yeah don't think there's any benefit to doing this atm. can re-open this issue in the future if that changes. see https://github.com/DetachHead/basedpyright/pull/202#issuecomment-2176087120
